assets/training/automl/environments/ai-ml-automl-dnn-text-gpu/context/Dockerfile

```diff
@@ -69,25 +69,26 @@ SHELL ["conda", "run", "-p", "$AZUREML_CONDA_ENVIRONMENT_PATH", "/bin/bash", "-c
 
 # begin pip install
 # Install pip dependencies
-# Here, we pin sentencepiece since 0.1.98 breaks training. Earlier versions of horovod contain a sev2 vulnerability,
-# and earlier versions of tokenizers cause log spam with transformers==4.16.0.
+# Install core packages first without conflicting dependencies
 RUN pip install \
                 # begin pypi dependencies
                 azureml-core=={{latest-pypi-version}} \
                 azureml-mlflow=={{latest-pypi-version}} \
                 azureml-automl-core=={{latest-pypi-version}} \
-                azureml-automl-dnn-nlp=={{latest-pypi-version}} \
                 azureml-automl-runtime=={{latest-pypi-version}} \
                 azureml-train-automl-client=={{latest-pypi-version}} \
                 azureml-train-automl-runtime=={{latest-pypi-version}} \
-                azureml-defaults=={{latest-pypi-version}} \
                 'azure-identity>=1.16.1' \
                 'sentencepiece==0.1.97' \
                 'cryptography>=42.0.5'\
                 'requests>=2.32.4' \
                 'certifi>=2023.07.22' 
                 # end pypi dependencies
 
+# Install conflicting packages separately with --no-deps
+RUN pip install 'azureml-defaults=={{latest-pypi-version}}' --no-deps --force-reinstall

```

Comment:
we should not try force install, if they are fixable in base packages. Could you add what conflict we are trying to avoid here?